TestDecreaseClusterSizeOf3: client: etcd cluster is unavailable or misconfigured

via https://jenkins-etcd-public.prod.coreos.systems/job/etcd-proxy/2048/console
```
--- FAIL: TestDecreaseClusterSizeOf3 (5.95s)
	cluster.go:295: client: etcd cluster is unavailable or misconfigured; error #0: client: etcd member unix://127.0.0.1:2269831860 has no leader
		; error #1: client: etcd member unix://127.0.0.1:2269631860 has no leader
```

--- a/integration/cluster.go
+++ b/integration/cluster.go
@@ -303,10 +303,11 @@ func (c *cluster) removeMember(t *testing.T, id uint64) error {
 	cc := MustNewHTTPClient(t, c.URLs(), c.cfg.ClientTLS)
 	ma := client.NewMembersAPI(cc)
 	ctx, cancel := context.WithTimeout(context.Background(), requestTimeout)
-	if err := ma.Remove(ctx, types.ID(id).String()); err != nil {
+	err := ma.Remove(ctx, types.ID(id).String())
+	cancel()
+	if err != nil {
 		return err
 	}
-	cancel()
 	newMembers := make([]*member, 0)
 	for _, m := range c.Members {
 		if uint64(m.s.ID()) != id {

--- a/integration/cluster_test.go
+++ b/integration/cluster_test.go
@@ -151,7 +151,15 @@ func testDecreaseClusterSize(t *testing.T, size int) {
 	// TODO: remove the last but one member
 	for i := 0; i < size-1; i++ {
 		id := c.Members[len(c.Members)-1].s.ID()
-		c.RemoveMember(t, uint64(id))
+		// may hit second leader election on slow machines
+		if err := c.removeMember(t, uint64(id)); err != nil {
+			if strings.Contains(err.Error(), "no leader") {
+				t.Logf("got leader error (%v)", err)
+				i--
+				continue
+			}
+			t.Fatal(err)
+		}
 		c.waitLeader(t, c.Members)
 	}
 	clusterMustProgress(t, c.Members)

